libgobuster/dir.go

```diff
@@ -59,6 +61,22 @@ func MakeRequest(s *State, fullUrl, cookie string) (*int, *int64) {
 		req.Header.Set("User-Agent", s.UserAgent)
 	}
 
+	if s.RandomUserAgent && s.UserAgent == "" {
+		b, err := ioutil.ReadFile(s.UserAgentsFile)
+		if err != nil {
+			return nil, nil, err
+		}
+		lines := strings.Split(string(b), "\n")
+		var uas []string
+		for _, line := range lines {
+			if !strings.HasPrefix(line, "#") && len(line) > 0 {
```

Comment:
Not 100%, but you may need to trim pre/post whitespace on here, otherwise a blank line with a few spaces might be counted as a valid UA.